composer.json

```diff
@@ -8,16 +8,19 @@
         "psr/http-client": "^1.0",
         "psr/http-message": "^1.0|^2.0",
         "psr/log": "^1.0|^2.0|^3.0",
-        "guzzlehttp/guzzle": "^7.5",
-        "guzzlehttp/psr7": "^2.0",
-        "webmozart/assert": "^1.11"
+        "webmozart/assert": "^1.11",
+        "psr/http-factory": "^1.0",
+        "php-http/discovery": "^1.19",
+        "psr/http-factory-implementation": "^1.0",
+        "psr/http-client-implementation": "^1.0"
```

Comment:
Same with other psr interfaces. Then we don't need `psr/http-client` - `psr/http-client-implementation` does a job

Reply:
I see. I inserted it to be able to use ClientInterface as an implementation and DI.